src/namespaces/asset_array.rb

```diff
@@ -15,6 +15,10 @@ def name
           @name ||= File.basename(path, '.yaml')
         end
 
+        def type
+          @type ||= File.basename(File.dirname(path).singularize)
```

Comment:
This looks to be fine however it looks a bit odd to `singularize` a file path. I can't say if that will work in all situations. A clearer approach would be:
```
File.basename(File.dirname(path)).singularize
```